File: ballonsmenu/src/main/java/com/karsu/ballonsmenu/Piece/Ham.java
package com.karsu.ballonsmenu.Piece;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.GradientDrawable;

import com.karsu.ballonsmenu.R;
import com.karsu.ballonsmenu.Util;

/**
 * Created by Erkan Kaplan on 2026-02-03
 */

final class Ham extends BoomPiece {

    public Ham(Context context) {
        super(context);
    }

    @Override
    public void init(int color, float cornerRadius) {
        Drawable backgroundDrawable = Util.getDrawable(this, R.drawable.piece, null);
        ((GradientDrawable)backgroundDrawable).setColor(color);
        ((GradientDrawable)backgroundDrawable).setCornerRadius(cornerRadius);
        Util.setDrawable(this, backgroundDrawable);
    }

    @Override
    public void setColor(int color) {
        ((GradientDrawable)getBackground()).setColor(color);
    }

    @Override
    public void setColorRes(int colorRes) { setColor(Util.getColor(getContext(), colorRes)); }
}

File: ballonsmenu/src/main/kotlin/com/karsu/ballonsmenu/Piece/Ham.kt
/** Created by Erkan Kaplan on 2026-02-03 */
package com.karsu.ballonsmenu.Piece

import android.content.Context
import android.graphics.drawable.GradientDrawable
import com.karsu.ballonsmenu.R
import com.karsu.ballonsmenu.Util

internal class Ham(context: Context) : BoomPiece(context) {

    override fun init(color: Int, cornerRadius: Float) {
        val backgroundDrawable = Util.getDrawable(this, R.drawable.piece, null)
        (backgroundDrawable as GradientDrawable).apply {
            setColor(color)
            this.cornerRadius = cornerRadius
        }
        Util.setDrawable(this, backgroundDrawable)
    }

    override fun setColor(color: Int) {
        (background as GradientDrawable).setColor(color)
    }

    override fun setColorRes(colorRes: Int) {
        setColor(Util.getColor(context, colorRes))
    }
}
